### PR TITLE
COS-7: Fix update sync information within contact editing process.

### PR DESCRIPTION
**Pull request by awg-github**
The  fields are getting updated when doing a update to the contact with inline edit from the contact summary page. 
cos-7_fix contact updating

### CODE REVIEW

File: odoosync.php [hunk 1]
```diff
@@ -143,12 +143,36 @@ function odoosync_civicrm_navigationMenu(&$menu) {
 /**
  * Implements hook_civicrm_postProcess().
  *
- * @throws \CiviCRM_API3_Exception
  */
 function odoosync_civicrm_postProcess($formName, &$form) {
-  $formAction = $form->getAction();
+  postProcessContactInlineForms($formName, $form);
```
Comment (omarabuhussein) [inline odoosync.php]: It think we can minimize the logic here and move it to the class as following  : 

```php
function odoosync_civicrm_postProcess($formName, &$form) {
  $syncInformationUpdater = new 
  CRM_Odoosync_Hook_PostProcess_ContactSyncInformationUpdater($formName, $form);
  $syncInformationUpdater->updateSyncInfo();
}



// now inside CRM_Odoosync_Hook_PostProcess_ContactSyncInformationUpdater class

public function __construct($formName, &$form) {
  $this->formName = $formName;
  $this->form = $form;
  $this->setContactIDs();
  $this->setSyncAction();
  // rest of the code
}

private function setContactIDs() {
  if (!empty($form->getVar('_contactIds')) {
    $this->contactIDs  = $form->getVar('_contactIds');
  } 
  else if (!empty($form->getVar('_contactId')) {
    $this->contactIDs  = [(int) $form->getVar('_contactId')];
  }
}

private function setSyncAction() {
  $isInlineContactForm = preg_match('/^CRM_Contact_Form_Inline_/', $formName);
  if ($isInlineContactForm && ($form->getAction() == CRM_Core_Action::UPDATE || 
  CRM_Core_Action::ADD || CRM_Core_Action::DELETE)) {
     $this->syncAction = 'update';
}
  
  if ($formName == "CRM_Contact_Form_Contact") {
  if ($form->getAction() == CRM_Core_Action::ADD) {
    $this->syncAction = 'create';
   }

if ($form->getAction() == CRM_Core_Action::UPDATE) {
  $this->syncAction = 'update';
 }
}
  
  if ($formName == "CRM_Contact_Form_Task_Delete" && $form->getAction() == 
  CRM_Core_Action::NONE) {
    $skipUnDelete = $form->getVar('_skipUndelete');
  if (!$skipUnDelete) {
  $this->syncAction = 'update';
}
}

public function updateSyncInfo() {
  if (empty($this->syncAction )) {
  return;
}

    $actionToSyncOptionValueId = $this->getOptionValueID('odoo_partner_action_to_sync', $this->syncAction );
    $param = [
      'custom_' . $this->syncStatusFieldId => $this->awaitingSyncOptionValueId,
      'custom_' . $this->actionToSyncFieldId => $actionToSyncOptionValueId,
      'custom_' . $this->actionDateFieldId => $this->currentDateTime,
    ];

foreach($this->contactIDs as $contactID) {
$param['contact_id'] = $contactID;
return civicrm_api3('Contact', 'create', $param);
}
}
```

File: odoosync.php [hunk 2]
```diff
@@ -143,12 +143,36 @@ function odoosync_civicrm_navigationMenu(&$menu) {
...
+}
+
+/**
+ * Updates contact sync information through editing contact form inline-blocks
+ *
+ * @param $formName
+ * @param $form
+ */
+function postProcessContactInlineForms($formName, $form) {
+  $isInlineContactForm = preg_match('/^CRM_Contact_Form_Inline_/', $formName);
```
Comment (omarabuhussein) [inline odoosync.php]: why not checking for the exact form name here ?

File: odoosync.php [hunk 3]
```diff
@@ -143,12 +143,36 @@ function odoosync_civicrm_navigationMenu(&$menu) {
...
+  if ($isInlineContactForm && ($form->getAction() == CRM_Core_Action::UPDATE || CRM_Core_Action::ADD || CRM_Core_Action::DELETE)) {
```
Comment (omarabuhussein) [inline odoosync.php]: but the only operation is **update** when you are doing inline edit, correct ?

**Review by omarabuhussein — CHANGES_REQUESTED** [review]

File: odoosync.php (hunk 3 above)
Comment (awg-github) [reply odoosync.php]: We can not only edit the data in contact inline forms.
For example, we can add new address, update it, and then delete it (I've tested all of them, functionality is separated into three different form actions)

**Review by omarabuhussein — APPROVED** [review]

**Comment by omarabuhussein** [comment]
approved @awg-github

**Pull request merged**